packages/gatsby/src/schema/node-model.js

```diff
@@ -703,18 +709,18 @@ const addRootNodeToInlineObject = (
   rootNodeMap,
   data,
   nodeId,
-  isNode = false,
-  path = new Set()
-) => {
+  isNode /*: boolean */,
+  path /*: Set<mixed> */
+) /*: void */ => {
+  if (path.has(data)) return
   path.add(data)
```

Comment:
Not caught in original PR, but could we move this to `if (isPlainObject || _.isArray(data)) {` block? We don't need to keep track of primitive values (numbers, strings, booleans, etc), so this might be slightly more performant (less additions to set / allocations)?